extensions/amp-smartlinks/0.1/test/test-amp-smartlinks.js

```diff
@@ -0,0 +1,246 @@
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *      http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS-IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+
+import * as DocumentReady from '../../../../src/document-ready';
+import * as LinkmateOptions from '../linkmate-options';
+import {AmpSmartlinks} from '../amp-smartlinks';
+import {LinkRewriterManager} from
+  '../../../amp-skimlinks/0.1/link-rewriter/link-rewriter-manager';
+import {Services} from '../../../../src/services';
+
+const helpersFactory = env => {
+  return {
+    createAmpSmartlinks(extensionAttrs) {
+      const ampTag = document.createElement('amp-smartlinks');
+
+      for (const attr in extensionAttrs) {
+        ampTag.setAttribute(attr, extensionAttrs[attr]);
+      }
+      ampTag.getAmpDoc = () => env.ampdoc;
+      return new AmpSmartlinks(ampTag);
+    },
+  };
+};
+
+describes.fakeWin('amp-smartlinks',
+    {amp: {extensions: ['amp-smartlinks']}},
+    env => {
+      let ampSmartlinks, helpers, xhr;
+
+      beforeEach(() => {
+        xhr = Services.xhrFor(env.win);
+        helpers = helpersFactory(env);
+        env.sandbox
+            .stub(DocumentReady, 'whenDocumentReady')
+            .returns(Promise.reject());
+      });
+
+      afterEach(() => {
+        env.sandbox.restore();
+      });
+
+      describe('getConfigOptions', () => {
+        it('Should parse options', () => {
+          env.sandbox.spy(LinkmateOptions, 'getConfigOptions');
+
+          const smartlinkOptions = {
+            'nrtv-account-name': 'thisisnotapublisher',
+            'linkmate': '',
+            'exclusive-links': '',
+            'link-attribute': 'href',
+            'link-selector': 'a',
+          };
+          ampSmartlinks = helpers.createAmpSmartlinks(smartlinkOptions);
+          env.sandbox.stub(ampSmartlinks, 'runSmartlinks_');
+
+          return ampSmartlinks.buildCallback().then(() => {
+            expect(LinkmateOptions.getConfigOptions.calledOnce).to.be
+                .true;
+            expect(ampSmartlinks.linkmateOptions_).to.deep.equal({
+              nrtvSlug: smartlinkOptions['nrtv-account-name'],
+              linkmateEnabled: true,
+              exclusiveLinks: true,
+              linkAttribute: smartlinkOptions['link-attribute'],
+              linkSelector: smartlinkOptions['link-selector'],
+            });
+          });
+        });
+
+        it('Should return handle bad options', () => {
+          env.sandbox.spy(LinkmateOptions, 'getConfigOptions');
+
+          const smartlinkOptions = {
+            'nrtv-account-name': 'alwaysastring',
+            'linkmate': 1234,
```

Comment:
nit: same

Reply:
We want this one here b/c it's a test asserting how we handle bad input. In this case we expect `linkmate` to still be true b/c we just check for it's presence in the tag.

Reply:
I see. That makes sense. I was curious why it was true before the validation change.